packages/grid/x-grid/src/tests/apiRef.XGrid.test.tsx

```diff
@@ -0,0 +1,77 @@
+import { ApiRef, useApiRef } from '@material-ui/data-grid';
+import { XGrid } from '@material-ui/x-grid/XGrid';
+import { expect } from 'chai';
+import * as React from 'react';
+import { getColumnValues, sleep } from 'test/utils/helperFn';
+import { render } from '@testing-library/react';
+
+describe('<XGrid /> - ApiRef', () => {
+  before(function beforeHook() {
+    if (/jsdom/.test(window.navigator.userAgent)) {
+      // Need layouting
+      this.skip();
+    }
+  });
+
+  const baselineProps = {
+    rows: [
+      {
+        id: 0,
+        brand: 'Nike',
+      },
+      {
+        id: 1,
+        brand: 'Adidas',
+      },
+      {
+        id: 2,
+        brand: 'Puma',
+      },
+    ],
+    columns: [{ field: 'brand' }],
+  };
+
+  let apiRef: ApiRef;
+
+  const TestCase = () => {
+    apiRef = useApiRef();
+    return (
+      <div style={{ width: 300, height: 300 }}>
+        <XGrid apiRef={apiRef} columns={baselineProps.columns} rows={baselineProps.rows} />
+      </div>
+    );
+  };
+
+  it('should allow to reset rows with setRows and render after 100ms', async () => {
+    render(<TestCase />);
+    const newRows = [
+      {
+        id: 3,
+        brand: 'Asics',
+      },
+    ];
+    apiRef.current.setRows(newRows);
+    await sleep(50);
+    expect(getColumnValues()).to.deep.equal(['Nike', 'Adidas', 'Puma']);
+    await sleep(50);
```

Comment:
We need tests to run fast and to be reliable. If you need to test performance, do it in a dedicated benchmark 

```suggestion
    act(() => {  
      apiRef.current.setRows(newRows);
    });
    expect(getColumnValues()).to.deep.equal(['Nike', 'Adidas', 'Puma']);
```

Reply:
Rows get updated every 100ms

Reply:
Maybe we could mock the time. Explicit sleep makes tests flaky and slow, best to avoid it.